### PR TITLE
fix: resolve t3:// links in inline linked images

### LINKED ISSUE #659: upgrade dev-main f081642 => dev-main 3bcfe1f
Hi, I'm glad that most of the issues #655 has been implemented or fixed. Only one point is still wrong. Inline-Img with links still uses Standalone.html, with unresolved links.

FE:
`<a class="image image-inline" href="t3://page?uid=1#1" target="_blank"></a>.`

A Question on minmal usage of linebreaks for Figcapiton: Would you please be so kind, to give me a hint on what you mean in #597 "Line breaks: Work naturally via text wrapping"? Whatever I do, linebreaks don't work. E.g. STRG+Enter or br in source-code is removed afer saving.

Another question about figcapitons: I noticed that 'german Umlauts' and some other special chars are not rendered like in tables. This can not be caused by css (font-family or something like that), because - like mentioned - in tables, figcaptions are as expected. 

Image

Image

### PR DESCRIPTION
**Pull request by CybotTM**
## Summary

Fixes #659. Inline images with links render with unresolved `t3://` URLs in the frontend because the `externalBlocks.a` configuration was dead code.

**Root cause**: TYPO3's `parseFunc` requires a two-part configuration for external blocks:
1. `externalBlocks = tag1, tag2, ...` — comma-separated list
2. `externalBlocks.tag1 { ... }` — per-tag configuration

Our extension added `externalBlocks.a { ... }` (part 2) but never added `a` to the list (part 1) via `:= addToList(a)`. Combined with `tags.a >` clearing the default handler, **all `<a>` tags passed through completely unprocessed**.

**Fix**: Replace the dead `externalBlocks.a` with a proper `tags.a` handler. Since `parseFunc` processes tags depth-first (inner first), `tags.img` fires before `tags.a` — so the new `renderInlineLink()` receives already-processed image content and only needs to:
1. Resolve `t3://` URLs via `typoLink_URL()`
2. Validate link protocols (security)
3. Reconstruct the `<a>` wrapper

## Changes

- **TypoScript**: Remove dead `externalBlocks.a` block, add `tags.a = TEXT` with `renderInlineLink`
- **ImageRenderingAdapter**: Add `renderInlineLink()` method + `extractLinkAttributes()` helper
- **Unit tests**: 10 new tests covering URL resolution, security, edge cases
- **Documentation**: Update method references

## Test plan

- [ ] Unit tests pass (`composer ci:test:php:unit` — 679 tests)
- [ ] PHPStan clean (`composer ci:test:php:phpstan`)
- [ ] E2E tests pass (CI)
- [ ] Inline linked image with `t3://page?uid=X` renders with resolved URL in frontend
- [ ] Block images with links in `<figure>` still work (handled by `renderFigure()`)
- [ ] `javascript:` protocol links are blocked (security)

### CODE REVIEW

File: Classes/Controller/ImageRenderingAdapter.php [hunk 1]
```diff
@@ -141,10 +141,66 @@ public function renderImageAttributes(?string $content, array $conf, ServerReque
...
+     *
+     * @see https://github.com/netresearch/t3x-rte_ckeditor_image/issues/659
+     */
+    #[AsAllowedCallable]
+    public function renderInlineLink(?string $content, array $conf, ServerRequestInterface $request): string
+    {
+        // Get inner content (already processed by tags.img via depth-first recursion)
+        $currentVal = $this->cObj instanceof ContentObjectRenderer
+            ? $this->cObj->getCurrentVal()
+            : null;
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Classes/Controller/ImageRenderingAdapter.php]: For `preUserFunc`, the `$content` parameter contains the **original** content before TypoScript processing, while `getCurrentVal()` returns the content **after** `current = 1` processing. Since `tags.a` is configured with `current = 1`, `getCurrentVal()` reflects the tag's inner content as parsed by `parseFunc` — this is the correct value to use.

This is consistent with the existing `renderImageAttributes()` method in the same class (line 113), which also uses `getCurrentVal()` rather than the `$content` parameter.

Resolving as by-design.
